cpp/src/aggregation/aggregation.cpp

```diff
@@ -428,6 +430,8 @@ std::unique_ptr<Base> make_product_aggregation()
 }
 template std::unique_ptr<aggregation> make_product_aggregation<aggregation>();
```

Comment:
Check if this could be deleted?
```suggestion
template<> std::unique_ptr<aggregation> make_product_aggregation<aggregation>() = delete;
```

Reply:
and similarly others.

Reply:
It can't.  See:  https://github.com/rapidsai/cudf/pull/10357#discussion_r815240117

Reply:
> How about making this constructor be pure virtual? (to make the class abstract).

Can't mark a constructor as virtual, no?